node/stateSync.go

```diff
@@ -171,10 +172,14 @@ func (s *stateSync) run() error {
 
 func (s *stateSync) startWorkerThread() {
 	var workerId uint32
-	peersNum := len(s.peers)
-	for workerId = 0; workerId < uint32(peersNum*concurrentSyncRequestPerNeighbor); workerId++ {
+	peersNum := uint32(len(s.peers))
+	workerNum := config.Parameters.SyncStateMaxPeer
+	if workerNum > peersNum {
+		workerNum = peersNum
+	}
+	for workerId = 0; workerId < workerNum*concurrentSyncRequestPerNeighbor; workerId++ {
```

Comment:
`concurrentSyncRequestPerNeighbor` should be moved to L175, otherwise we will have more workers than `config.Parameters.SyncStateMaxPeer`

Reply:
This is still not fixed yet

Reply:
Fixed